app/models/concerns/hyrax/work_behavior.rb

```diff
@@ -30,6 +30,13 @@ module WorkBehavior
       self.indexer = WorkIndexer
       # Default VirusScanner, configurable for Hyrax work types
       self.default_system_virus_scanner = Hyrax::VirusScanner
+      after_update :conditionally_update_embargo_and_lease
+    end
+
+    def conditionally_update_embargo_and_lease
+      return true unless visibility_changed?
```

Comment:
is this correct? can't i change, e.g., an embargo release date without changing object visibility?

Reply:
I'll see if `embargo.changed?` and `lease.changed?` can work.